lib/src/wallet/keys/unified.rs

```diff
@@ -0,0 +1,406 @@
+use std::{
+    collections::{HashMap, HashSet},
+    io::{self, Read, Write},
+};
+
+use bip0039::Mnemonic;
+use byteorder::{ReadBytesExt, WriteBytesExt};
+use orchard::keys::Scope;
+
+use zcash_client_backend::address::UnifiedAddress;
+use zcash_encoding::Vector;
+use zcash_primitives::{legacy::TransparentAddress, zip32::DiversifierIndex};
+use zingoconfig::ZingoConfig;
+
+use crate::wallet::traits::ReadableWriteable;
+
+use super::{extended_transparent::KeyIndex, get_zaddr_from_bip39seed, ToBase58Check};
+
+#[derive(Clone, Debug)]
+pub struct UnifiedSpendAuthority {
+    orchard_key: orchard::keys::SpendingKey,
+    sapling_key: zcash_primitives::zip32::ExtendedSpendingKey,
+    transparent_parent_key: super::extended_transparent::ExtendedPrivKey,
+    transparent_child_keys: Vec<(usize, secp256k1::SecretKey)>,
+
+    addresses: Vec<UnifiedAddress>,
+    // Not all diversifier indexes produce valid sapling addresses.
+    // Because of this, the index isn't necessarily equal to addresses.len()
+    next_sapling_diversifier_index: DiversifierIndex,
+
+    // Note that unified spend authority encryption is not yet implemented,
+    // These are placeholder fields, and are currenly always false
+    pub(crate) encrypted: bool,
+    pub(crate) unlocked: bool,
+}
+
+#[derive(Debug, Clone, Copy)]
+#[non_exhaustive]
```

Comment:
According to [the release notes](https://blog.rust-lang.org/2019/12/19/Rust-1.40.0.html#non_exhaustive-structs-enums-and-variants) the rationale for the [`#[non_exhaustive] attribute`](https://doc.rust-lang.org/reference/attributes.html)

Reply:
Yep! It's a drop in the bucket when it comes to the amount of breaking changes a new pool would likely entail for our public interface, but this means consumers of a `ReceiverSelection` would need to have a catch-all match for future receiver types.